k8s/eks/bash/functions.sh

```diff
@@ -0,0 +1,312 @@
+  echo "$create_ebs_volume"
+  ebs_volume=$(echo $create_ebs_volume | jq -r '.VolumeId' )
+  echo "ebs=$ebs_volume" >> ./.env
+}
+
+make_persistent_volume(){  
+export TG_EBS_DATADIR_VOLUME_ID=$ebs_volume
+EBS_PV=$(mktemp)
+envsubst <../kops/ebs/pv.yml.spec >$EBS_PV
+kubectl apply -f ../kops/ebs/storageclass.yml -f $EBS_PV -f ../kops/ebs/pvc.yml
+}
+
+helm_redis_add_repo(){
+  helm repo add bitnami https://charts.bitnami.com/bitnami
+} 
+
+helm_infra_install_redis(){
+  helm install testground-infra-redis --set auth.enabled=false --set master.nodeSelector='testground.node.role.infra: "true"' bitnami/redis
+} 
+
+# Until we find a better way for region var
+# tg_daemon_config_map(){
+#   kubectl apply -f ../kops/testground-daemon/config-map-env-toml.yml
+# } | tee -a ./log/$start-log/tg_daemon_config_map.log
+
+tg_daemon_config_map(){
+  kubectl create -f - <<EOF
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: env-toml-cfg
+  namespace: default
+data:
+  .env.toml: |
+    ["aws"]
+    region = "$REGION"   
+
+    [runners."cluster:k8s"]
+    run_timeout_min             = 15
+    testplan_pod_cpu            = "100m"
+    testplan_pod_memory         = "100Mi"
+    collect_outputs_pod_cpu     = "1000m"
+    collect_outputs_pod_memory  = "1000Mi"
+    provider                    = "aws"
+    sysctls = [
+      "net.core.somaxconn=10000",
+    ]
+
+    [daemon]
+    listen = "0.0.0.0:8042"
+    slack_webhook_url = ""
+    github_repo_status_token = ""
+
+    [daemon.scheduler]
+    workers = 2
+    task_timeout_min  = 20
+    task_repo_type    = "disk"
+
+    [client]
+    endpoint = "localhost:8080"
+
+EOF
+}
+
+tg_daemon_service_account(){
+  kubectl apply -f ./yaml/tg-daemon-service-account.yml
+}
+
+tg_daemon_role_binding(){
+  kubectl apply -f ./yaml/tg-daemon-cluster-role.yml
+}
+
+tg_daemon_services(){
+  kubectl apply -f ./yaml/tg-testground-daemon-service.yml
+}
+
+tg_daemon_svc_sync_service(){
+  kubectl apply -f ./yaml/tg-sync-service.yml
+}
+
+tg_daemon_sync_service(){
+  kubectl apply -f ./yaml/tg-sync-service-deployment.yml
+}
+
+tg_daemon_sidecar(){
+  kubectl apply -f ./yaml/tg-ds-sidecar.yml
+}
+
+tg_daemon_deployment(){
+  kubectl apply -f ./yaml/tg-daemon-deployment.yml
+}
+
+log(){
+  tar czf ./log/$start-$CLUSTER_NAME-$CNI_COMBINATION.tar.gz /log/$start-log/
+  echo "##########################################"
+  echo "Log file generated with name $start-$CLUSTER_NAME-$CNI_COMBINATION.tar.gz"
+  echo "##########################################"
+  rm -rf ./log/$start-log/ 
+}
+
+# cluster_creation_manifest(){
+#   mkdir -p .cluster_manifest
+#   echo "$CLUSTER_NAME" >> clusters
+# }
+
+cleanup(){
+  mnt_target_id=$(aws efs describe-mount-targets --file-system-id $efs | jq -r ".MountTargets[] | .MountTargetId")
+  echo "Removing mount target with ID $mnt_target_id"
+  echo "The script will pause here for 20 seconds for the removal to complete"
+  aws efs delete-mount-target --mount-target-id $mnt_target_id
+  sleep 20
```

Comment:
I think AWS provides a mechanism exactly for that, we will look into it.

Reply:
Why don't you use `aws efs describe-file-systems` or whatever else can give you a status of EFS to check if EFS is actually deleted instead of sleeping for a fixed amount of time? We don't have to wait for Amazon to implement `--wait` arg. We can implement that ourselves. And that's the ask here.

Reply:
This has been updated.